cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -489,19 +529,32 @@ private boolean isLeadershipMovementDone(Cluster cluster, TopicPartition tp, Exe
     private boolean maybeMarkTaskAsDeadOrAborting(Cluster cluster, ExecutionTask task) {
       // Only check tasks with IN_PROGRESS or ABORTING state.
       if (task.state() == IN_PROGRESS || task.state() == ABORTING) {
-        if (task.type() == ExecutionTask.TaskType.LEADER_ACTION
-            && cluster.nodeById(task.proposal().newLeader()) == null) {
-          _executionTaskManager.markTaskDead(task);
-          LOG.warn("Killing execution for task {} because the target leader is down", task);
-          return true;
-        } else {
-          for (int broker : task.proposal().newReplicas()) {
-            if (cluster.nodeById(broker) == null) {
+        switch (task.type()) {
```

Comment:
The `JavaDoc` of this function does not describe the new behavior (code changes).